src/main/java/frc/robot/Systems/Chassis/Chassis.java

```diff
@@ -109,12 +140,24 @@ public void resetPose(Pose2d pose) {
 
   @Override
   public void periodic() {
+    PhotonPipelineResult result = ATCamera.getLatestResult();
+    Optional<EstimatedRobotPose> estimatedPose = getEstimatedGlobalPose(getFusedPose());
+
+    if(estimatedPose.isPresent()) {
+      poseEstimator.addVisionMeasurement(estimatedPose.get().estimatedPose.toPose2d(), estimatedPose.get().timestampSeconds);
+      Logger.recordOutput("AT Estimate", estimatedPose.get().estimatedPose.toPose2d());
+    }
     poseEstimator.update(Rotation2d.fromDegrees(getGyroRot()), getPositions()); //TODO this gyro angle might have to be negated
 
     SmartDashboard.putNumber("gyroYaw", getGyroRot());
     Logger.recordOutput("Estimated Pose", poseEstimator.getEstimatedPosition());
   }
 
+   public Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedRobotPose) {
+        photonEstimator.setReferencePose(prevEstimatedRobotPose);
```

Comment:
since the strategy for pose estimation is `MUTI_TAG_PNP_ON_COPROCESSOR` the reference pose remains unused. I would remove this method and just access the result of `photonEstimator.update()` in `periodic()`